charts/sourcegraph/templates/otel-collector/otel-collector.Deployment.yaml

```diff
@@ -0,0 +1,121 @@
+{{- if .Values.openTelemetry.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    description: Receives, processes, and exports telemetry data.
+  labels:
+    {{- include "sourcegraph.labels" . | nindent 4 }}
+    {{- if .Values.openTelemetry.gateway.labels }}
+      {{- toYaml .Values.openTelemetry.gateway.labels | nindent 4 }}
+    {{- end }}
+    deploy: sourcegraph
+    app.kubernetes.io/component: otel-collector
+  name: {{ .Values.openTelemetry.gateway.name }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "sourcegraph.selectorLabels" . | nindent 6 }}
+      app: otel-collector
+  minReadySeconds: 5
+  progressDeadlineSeconds: 120
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: otel-collector
+      {{- if .Values.sourcegraph.podAnnotations }}
+      {{- toYaml .Values.sourcegraph.podAnnotations | nindent 8 }}
+      {{- end }}
+      {{- if .Values.openTelemetry.gateway.podAnnotations }}
+      {{- toYaml .Values.openTelemetry.gateway.podAnnotations | nindent 8 }}
+      {{- end }}
+      labels:
+      {{- include "sourcegraph.selectorLabels" . | nindent 8 }}
+      {{- if .Values.sourcegraph.podLabels }}
+      {{- toYaml .Values.sourcegraph.podLabels | nindent 8 }}
+      {{- end }}
+      {{- if .Values.openTelemetry.gateway.podLabels }}
+      {{- toYaml .Values.openTelemetry.gateway.podLabels | nindent 8 }}
+      {{- end }}
+        deploy: sourcegraph
+        app: otel-collector
+    spec:
+      containers:
+      - name: otel-collector
+        image: {{ include "sourcegraph.image" (list . "openTelemetry") }}
+        imagePullPolicy: {{ .Values.sourcegraph.image.pullPolicy }}
+        command:
+          - "/bin/otelcol-sourcegraph"
+          {{- if and (not .Values.openTelemetry.gateway.config.traces.exporters) (not .Values.jaeger.enabled) }}  # default config: enable logging exporter
+          - "--config=/etc/otel-collector/configs/logging.yaml"
+          {{- else if .Values.jaeger.enabled }} # use bundled config to connect to bundled jaeger
+          - "--config=/etc/otel-collector/configs/jaeger.yaml"
+          {{- else }} # use config defined by customer in values.yaml
+          - "--config=/etc/otel-collector/config.yaml"
+          {{- end -}}
```

Comment:
very nice

Reply:
we might want to make it clear if you use the bundled Jaeger you cannot add your own exporters

Reply:
there's an error here: https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/167/files#diff-2c97c607aab6b149c3c126e2b863f3654bbf1fac26187ff909a735ca35c88ca0R2